src/common/debugAdapter.ts

```diff
@@ -292,7 +297,10 @@ export class DebugAdapter implements vscode.DebugAdapter {
 		return this._outputChain;
 	}
 
-	private _translateToWorkspacePath(wasmPath: string): string {
+	private async _translateToWorkspacePath(wasmPath: string): Promise<string> {
+		// Have to wait for path mappings to come in from the device
+		await this._pathMappingsComplete;
```

Comment:
Root cause of the path mappings not working is they weren't set until after the process already finished launching. Breakpoint requests come in before that.

Reply:
This might screw up stopOnEntry : false because the breakpoints won't have a chance to be bound before the process starts running.

Reply:
Could pathMappings be changed to a request? That would fix the problem. I'll try that instead. Then the request could be awaited.

Reply:
Okay it's not the notification that's the problem, it's the waiting for it in the adapter. Never mind about switching the notification to a request.

Reply:
And stopOnEntry still works because it waits for configuration done to continue. The initial breakpoints do get a chance to be created.